docs/release-notes.md

```diff
@@ -10,6 +10,90 @@ toc_max_heading_level: 2
+  - Extract mode now has a concatenation reduction (`concat`). Disjoint snippets within a file can be treated as one unified file. 
+  - You can use extract mode to scan for generic languages (use value `generic` in `dest-language`).
+
+- Taint mode:
+  - Add experimental support for _taint labels_, which is an ability to attach labels to different kinds of taint. Both sources and sinks can restrict what labels are present in the data that passes through them in order to apply. This allows you to write more complex taint rules that previously required unappealing workarounds. Taint labels are also helpful for writing certain classes of typestate analyses (for example, check that a file descriptor is not used after being closed).
+  - Introduced the `--dataflow-traces` flag, which directs the Semgrep CLI to explain how non-local values lead to a finding. Currently, this only applies to taint mode findings and it traces the path from the taint source to the taint sink.
+  - Added taint traces as part of Semgrep JSON output. This helps explain how the sink became tainted.
+
+- General and language support additons:
+  - New language Elixir with experimental support!
+  - Scala: Ellipsis are now allowed in for loop function headers, allowing you to write patterns such as `for (...; $X <- $Y if $COND; ...) { ... }` to match nested for loops. (Issue [#5650](https://github.com/returntocorp/semgrep/issues/5650))
+  - Kotlin: Support for ellipsis in field access (for example, `obj. ... .bar()`).
+  - Semgrep now reports, among other metadata, file extensions from App-connected scans that do **not** match the language of any enabled rule. This addition can make the development of new rules more effective by improving language prioritization.
+  - Previously, expression statement patterns (for example `foo();`) were always matching when the expression statement was a bit deeper in the expression (for example, `x = foo();`). This default behavior can now be disabled through rule `options:` with `implicit_deep_exprstmt: false` in rules YAML file. (Issue [#5472](https://github.com/returntocorp/semgrep/issues/5472))
+  - Changed `semgrep-core` so it runs on YAML files that do not have a top-level `rules: ...` key. As a result, you can now copy from the playground editor directly into a local rules YAML file for use with `semgrep-core`!
+  - LSP support: Improving **experimental** Language Server Protocol (LSP) support for metavariable inlay hints, hot reloading, App integration, scan commands, and much more!
+
+#### Changes
+
+- Breaking changes in the `dataflow_trace` JSON output to make it more easily consumable by Semgrep App. Added content for `taint_source` and `intermediate_vars`, and collapsed the multiple `taint_source` locations into one.
+
+- General performance improvements:
+  - By default, Semgrep no longer stores the time or output of skipped targets. This improvement significantly reduced Semgrep's memory consumption in large repositories!
```

Comment:
Is the audience of release notes interested in the inner details of `Semgrep no longer stores the time or output of skipped targets` ?
Should we just remove these details?

Reply:
I think this might be still interesting, because of the result.

Reply:
```suggestion
  - Semgrep significantly reduced its memory consumption in large repositories!
```